velox/docs/stats.rst

```diff
@@ -0,0 +1,126 @@
+==============
+Stats Counters
+==============
+Stats counter is used to collect the metrics of important velox runtime events
+for monitoring purpose. The collected metrics can provide insights into the
+continuous availability and performance analysis of a velox runtime system. For
+instance, the collected data can help automatically generate alerts at an
+outage. Velox provide a framework to collect the customized stats which consists
+of three steps:
+
+**Stats Definition**: define the name, type and config parameters for each stats,
+and register through registerVeloxCounters() on velox runtime startup.
+
+**Stats Record**: when the event of a predefined stats happens in production, the
+velox runtime records the stats data point through the stats collection
+interface provided by BaseStatsReporter.
+
+**Stats Export**: BaseStatsReporter aggregates the collected data points based
+on the registered stats definition, and periodically exports to the backend
+monitoring service. The implementation of BaseStatsReporter is based on the
```

Comment:
> The implementation of BaseStatsReporter is based on the
actually integrated monitoring service

Isn't BaseStatsReporter just a default implementation that does nothing? Shouldn't the developer create their own implementation derived from BaseStatsReporter and register it somehow?